Laravelsix

Patabugen opened the pull request:
Here's a whole bunch more additions! See the git-messages or changelog.md for details.

I think you still need to manage releases/version numbers, I'm not sure I can do that from my end, by the way.

This is a bit of a messy update - with lots of things merged together. I'll try and calm things down and have feature branches and more specific pull requests in future. I think this updates it enough for my current use case - however - so there may not be many more.

Commit:
Moved Route specific methods to the Route Response

Commit:
Added some error handling, added base Route object for responses, renamed DrivingDistance test.

Commit:
Removed ExampleTest

Commit:
Added getDistance and fixed bug in RouteServiceResponse

Commit:
Added error handling to Requests and fixed potential bug in URL generation.

Commit:
Renamed ResponseJson to ResponseData (it isn't JSON). Added status codes and methods to AbstractResponse.

Commit:
Fixed bugs in RouteServiceResponse

Commit:
Setup testing config defaults to use Demo server. Changed demo coordinates to use Berlin (as used in Quick Start demo for custom servers).

Commit:
Added NoRoute status code and error handling on Get Distance.

Commit:
Corrected distance checking type

Commit:
Created LatLng model to disambiguate the order of LatLngs. Changed distance test coordinates to avoid traffic anomolies.

Commit:
Corrected default distance unit and corrected tests.

Commit:
Updated readme and changelog, and fixed test.

Commit:
Updated readme.md and added phpunit.xml to gitignore

Commit:
Added documentation and getRotueData() for raw data access

dmgctrlr commented:
Looks good at a first once over. I'll get us a server setup this evening so we have something to test against.

Patabugen commented:
The tests work against the demo server - but I frequently get a 429 error (Too Many Requests) - which I think is the server being busy (rather than me making too many requests). So I often need to run them a few times.

I've done a bunch of fiddling to get an OSRM docker server running locally, if you need any help doing that.

dmgctrlr commented:
I saw that 429 alright. If you have something running locally, great! We
have an internal server which I can test against, when I get some time!
Thanks again for the PR

On Tue, Nov 5, 2019 at 11:42 AM Sami Greenbury <notifications@github.com>
wrote:

> The tests work against the demo server - but I frequently get a 429 error
> (Too Many Requests) - which I think is the server being busy (rather than
> me making too many requests). So I often need to run them a few times.
>
> I've done a bunch of fiddling to get an OSRM docker server running
> locally, if you need any help doing that.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dmgctrlr/lara-osrm/pull/5?email_source=notifications&email_token=AA56LCXJXVNMARZJGF3SAETQSG5A7A5CNFSM4JJF7AE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEDD4AVY#issuecomment-549961815>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA56LCVZ4OLS5BJTIYOQCGLQSG5A7ANCNFSM4JJF7AEQ>
> .
>

Patabugen commented:
I think I've broken the Facade bit of this, partly because I don't entirely understand that bit of Laravel!

I can't get the original example code to run:

`$response = LaraOsrmFacade::drivingDistance();`

Patabugen commented:
Welcome!

I also think I've broken the non Route services - because I've not implemented any other Response objects. So a few steps forward and a few steps backwards!

Commit:
Updated changelog with details of breaking changes, added Request section to Readme.md

Patabugen commented:
I've figured out the Facades bit and added to the documentation. I also added notes on my breaking changes.

https://github.com/dmgctrlr/lara-osrm/pull/5/commits/32929e5b9e70100615f9fd274744fe5a050c3055

Commit:
Added basic support for TripRequest, and fixed a couple of bugs.

Commit:
Fixed method for checking if distance is present

Commit:
Separated URL generation into it's own function.

Patabugen commented:
We have Version 1!

https://github.com/dmgctrlr/lara-osrm/tree/1.0

dmgctrlr commented:
Oh wow, great work! Very cool to have v1 at last! I think I might be
needing this soon in my work, so when that happens I'll be able to devote a
lot more time to it.

Nice work!

On Fri, Nov 22, 2019 at 5:21 AM Sami Greenbury <notifications@github.com>
wrote:

> We have Version 1!
>
> https://github.com/dmgctrlr/lara-osrm/tree/1.0
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dmgctrlr/lara-osrm/pull/5?email_source=notifications&email_token=AA56LCSFKR27GED6F3IH55DQU7FD5A5CNFSM4JJF7AE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEE5PKIA#issuecomment-557511968>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA56LCQYTCWT65ERS6MONN3QU7FD5ANCNFSM4JJF7AEQ>
> .
>

Patabugen commented:
Thanks! It's working as much as I need it to now - though I'm aware I've not fully implemented everything I've broken realistically I'm unlikely to do anything else without either needing it or being prodded to - so feel free to ask/chat when you do start needing it.
